driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -86,38 +96,59 @@ public void run() {
             }
         };
 
+        this.connections = new CopyOnWriteArrayList<Connection>();
+        this.open = new AtomicInteger();
+
         // Create initial core connections
-        List<PooledConnection> l = new ArrayList<PooledConnection>(options().getCoreConnectionsPerHost(hostDistance));
-        try {
-            for (int i = 0; i < options().getCoreConnectionsPerHost(hostDistance); i++)
-                l.add(manager.connectionFactory().open(this));
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            // If asked to interrupt, we can skip opening core connections, the pool will still work.
-            // But we ignore otherwise cause I'm not sure we can do much better currently.
-        } catch (ConnectionException e) {
-            forceClose(l);
-            throw e;
-        } catch (UnsupportedProtocolVersionException e) {
-            forceClose(l);
-            throw e;
-        } catch (ClusterNameMismatchException e) {
-            forceClose(l);
-            throw e;
-        } catch (RuntimeException e) {
-            forceClose(l);
-            throw e;
+        final List<ListenableFuture<Connection>> connectionFutures =
+            Lists.newArrayListWithCapacity(options().getCoreConnectionsPerHost(hostDistance));
+        for (int i = 0; i < options().getCoreConnectionsPerHost(hostDistance); i++) {
+            if (preExistentConnection != null && preExistentConnection.setPool(this)) {
+                SettableFuture<Connection> future = SettableFuture.create();
+                future.set(preExistentConnection);
+                connectionFutures.add(future);
```

Comment:
The compareAndSet in setPool handles this, but it's not very explicit so I wouldn't mind adding what you suggested.